public/app/features/explore/extensions/AddToDashboard/addToDashboard.ts

```diff
@@ -28,14 +30,77 @@ function createDashboard(): DashboardDTO {
   return dto;
 }
 
+/**
+ * Returns transformations for the logs table visualisation in explore.
+ * If the logs table supports a labels column, we need to extract the fields.
+ * Then we can set the columns to show in the table via the organize/includeByName transformation
+ * @param panelType
+ * @param options
+ */
+function getLogsTableTransformations(panelType: string, options: AddPanelToDashboardOptions): DataTransformerConfig[] {
+  let transformations: DataTransformerConfig[] = [];
+  if (panelType === 'table' && options.panelState?.logs?.columns) {
+    // If we have a labels column, we need to extract the fields from it
+    if (options.panelState.logs?.labelName) {
+      transformations.push({
+        id: 'extractFields',
+        options: {
+          source: options.panelState.logs.labelName,
+        },
+      });
+    }
+
+    // Show the columns that the user selected in explore
+    transformations.push({
+      id: 'organize',
+      options: {
+        includeByName: Object.values(options.panelState.logs.columns).reduce(
+          (acc: Record<string, boolean>, value: string) => ({
+            ...acc,
+            [value]: true,
+          }),
+          {}
+        ),
+      },
+    });
+  }
+  return transformations;
+}
+
+/**
+ * Returns the transformations to apply to the panel based on the panel type
+ * @param panelType
+ * @param options
+ */
+function getExploreToPanelTransformations(panelType: string, options: AddPanelToDashboardOptions) {
+  let transformations: DataTransformerConfig[] = [];
+  transformations.push(...getLogsTableTransformations(panelType, options));
+
+  return transformations;
+}
+
+/**
+ * Filters the query for logs table, if the user has a query selected in explore, only return that one
+ * The logs table uses organize fields transformation, which only supports a single dataframe
+ * Also, the UI only displays one dataframe/query at a time, so this should result in the expected behavior.
+ * @param options
+ */
+function getExploreToPanelQueries(options: AddPanelToDashboardOptions) {
+  return options?.panelState?.logs?.refId
+    ? [options.queries.find((q) => q.refId === options?.panelState?.logs?.refId)]
+    : options.queries;
+}
```

Comment:
I'm not sure about this, if the user has a a metric query (or any other result type) and a logs query, the metrics query is completely ignored. i think the source of the issue here is that e2d adds everything to a single panel, and this doesn't really scale well in this scenario

Reply:
Yeah, so I guess we can either pull everything into the new panel, just pull the first query into the panel, or create a panel for each.

But on main I'm seeing this same issue, when creating multiple queries, the query type of the first query defines the panel type, and all other queries in explore are dumped into the same dashboard panel.

I would argue that this is a pre-existing issue that isn't introduced in this PR? What do you think @Elfo404?

Reply:
the problem here though is that we are removing queries, while on main what happens is that all the queries are pulled into the panel, and the panel itself is - as you said - picked from the available ones based on the result.

we probably need to do something else in explore, and rethink a bit how the "add to dashboard" works, possibly being a bit smarter  about the number of panels/queries.

cc @grafana/explore-squad

Reply:
i'd say that would be the best thing to do for now as it is the documented behaviour

Reply:
Should be pushed up now